app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaCountryPickerPresenter.kt

```diff
@@ -0,0 +1,69 @@
+package org.p2p.wallet.striga.ui.countrypicker
+
+import kotlinx.coroutines.launch
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.repository.Country
+import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.striga.onboarding.interactor.StrigaOnboardingInteractor
+import org.p2p.wallet.striga.ui.countrypicker.delegates.StrigaCountryCellModel
+import org.p2p.wallet.striga.ui.countrypicker.delegates.StrigaCountryHeaderCellModel
+import org.p2p.wallet.utils.emptyString
+
+class StrigaCountryPickerPresenter(
+    private val strigaOnboardingInteractor: StrigaOnboardingInteractor,
+    private val selectedCountry: Country? = null
+) : BasePresenter<StrigaCountryPickerContract.View>(),
+    StrigaCountryPickerContract.Presenter {
+
+    private var searchText: String = emptyString()
+    private val searchTextMap = hashMapOf<String, List<Country>>()
+    private var allCountryCodeItems: List<Country> = listOf()
+
+    override fun search(text: String) {
+        searchText = text
+        searchByCountryName(text)
+    }
+
+    override fun attach(view: StrigaCountryPickerContract.View) {
+        super.attach(view)
+        launch {
+            allCountryCodeItems = strigaOnboardingInteractor.getAllCountries()
+            val mappedItems = mapToCellItem(allCountryCodeItems)
+            view.showCountries(buildCellList(mappedItems))
+        }
+    }
+
+    private fun searchByCountryName(countryName: String) {
+        launch {
+            when (countryName) {
+                in searchTextMap -> {
+                    val cachedItems = mapToCellItem(searchTextMap[countryName].orEmpty())
```

Comment:
maybe you can keep ready StrigaCountryCellModel in map ?

Reply:
I think it's not good idea, we should discuss about this, what we should do with ui items, when we need domain models to be mapped before they go to adapter